planning/behavior_path_planner/src/utils/utils.cpp

```diff
@@ -2418,10 +2419,22 @@ PathWithLaneId getCenterLinePath(
     const double forward_length = std::max(goal_arc_coordinates.length - lane_change_buffer, 0.0);
     s_forward = std::min(s_forward, forward_length);
   }
+#else
+  if (route_handler.isDeadEndLanelet(lanelet_sequence.back())) {
+    const auto lane_length = lanelet::utils::getLaneletLength2d(lanelet_sequence);
+    s_forward = std::clamp(s_forward, 0.0, lane_length);
+  }
+
+  if (route_handler.isInGoalRouteSection(lanelet_sequence.back())) {
+    const auto goal_arc_coordinates =
+      lanelet::utils::getArcCoordinates(lanelet_sequence, route_handler.getGoalPose());
+    s_forward = std::clamp(s_forward, 0.0, goal_arc_coordinates.length);
+  }
+#endif
 
   const auto raw_path_with_lane_id =
     route_handler.getCenterLinePath(lanelet_sequence, s_backward, s_forward, true);
-  const auto resampled_path_with_lane_id = motion_utils::resamplePath(
+  auto resampled_path_with_lane_id = motion_utils::resamplePath(
```

Comment:
Is it okay not to remove `const` ?

Reply:
I removed it due to `clang-tidy` is giving warning message 
** Constness of 'resampled_path_with_lane_id' prevents automatic move (lsp)**

I can add them back if needed.

Reply:
Thanks for the information!